qannotate/src/au/edu/qimr/qannotate/Main.java

```diff
@@ -16,58 +16,105 @@
+            } else if (options.getMode() == Options.MODE.make_valid) {
+    	   		new MakeValidMode( options );
 //           } else if (options.getMode() == Options.MODE.snppileup) {
 //   	   			new SnpPileupMode( options );
-   	   	   } else if (options.getMode() == Options.MODE.overlap) {
+   	   	    } else if (options.getMode() == Options.MODE.overlap) {
    	   			new OverlapMode( options );
 	   	   	} else if (options.getMode() == Options.MODE.vcf2maftmp) {
 				new Vcf2mafTmp(  options );
-           } else {
-        	   		logger.error("No valid mode are specified on commandline: " + options.getMode());
-        	   		throw new Exception("No valid mode are specified on commandline: " + options.getMode()) ;
-           }
+	   	   	} else if (options.getMode() == null) {
+	   	   		throw new IllegalArgumentException("No mode was specified on the commandline - please add the \"-mode\" option") ;
+            } else {
+    	   		throw new IllegalArgumentException("No valid mode are specified on commandline - please run \"qannotate -help\" to see the list of available modes");
+            }
 
             logger.logFinalExecutionStats(0);
                
         } catch (Exception e) {
-	        	System.out.println("Exception caught!");
-	        	e.printStackTrace();
-	        	System.err.println(Thread.currentThread().getName() + " " + e.toString() + " : " + e.getLocalizedMessage());
-	        	if (null != logger) {
-		        	logger.info(Thread.currentThread().getName() + " " + e.toString() + " : " + e.getMessage());	            
-		        	logger.logFinalExecutionStats(1);
-	        	}
-        		System.out.println("About to return exit code of 1");
-            System.exit(1);
-        }			
-		
-		
+        	System.out.println("Exception caught!");
+        	e.printStackTrace();
+        	System.err.println(Thread.currentThread().getName() + " " + e.toString() + " : " + e.getLocalizedMessage());
+        	if (null != logger) {
+	        	logger.info(Thread.currentThread().getName() + " " + e.toString() + " : " + e.getMessage());	            
+	        	logger.logFinalExecutionStats(1);
+        	}
+    		System.out.println("About to return exit code of 1");
+    		System.exit(1);
+        }
+	}
+	
+	
+	public static void checkOptions(Options o) {
+		Options.MODE m = o.getMode();
+		switch (m) {
+		/*
+		 * some modes need a database
+		 */
+		case dbsnp:
+		case germline:
+		case hom:
+		case cadd:
+		case snpeff:
+		case trf:
+			if (null == o.getDatabaseFileName()) {
+        		throw new IllegalArgumentException("Please supply a reference file using the \"-d\" option");
+        	}
+			
+		/*
+		 * modst modes need an output
+		 */
+		case overlap:
+		case confidence:
+		case indelconfidence:
+		case ccm:
+		case make_valid:
+			if (null == o.getOutputFileName()) {
+				throw new IllegalArgumentException("Please supply an output file using the \"-output\" or \"-o\" option");
+			}
+			if (null == o.getInputFileName()) {
```

Comment:
is this check for input repeated below?

Reply:
It is, but the `vcf2maf` modes can take either `output` or `outdir` as options, whereas the other modes need `output`
I'll add some comments